### PR TITLE
Add color inversion OSL shader

### PR DESCRIPTION
**Pull request by Mango-3**
This shader is an OSL implementation of RenderMan's PxrInvert node:

https://rmanwiki.pixar.com/display/REN22/PxrInvert

It allows to invert individual or all color channels of an input color with selectable color model. Output is a RGB color.

### CODE REVIEW

File: src/appleseed.shaders/src/appleseed/as_inv_color.osl [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    string as_category = "color",
+    
+    string as_max_class_id = "1105087319 977155459", 
+    string as_max_plugin_type = "texture",
+    
+    int as_maya_type_id = 0x00127a08,
+    string as_maya_classification = "drawdb/shader:rendernode/appleseed/utility"
+]]
+(
+    color in_color = color(0,0,0)
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_inv_color.osl]: Insert spaces after commas.

File: src/appleseed.shaders/src/appleseed/as_inv_color.osl [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        string as_maya_attribute_short_name = "oc",
+        string widget = "null",
+        string label = "Color"
+    ]]
+)
+{
+    color in_color_trans = transformc(in_color_model, in_color);
+    color out_color_trans = color(0);
+    
+    if (as_invert_channel_0 == 1 && as_invert_channel_1 == 0 && as_invert_channel_2 == 0)
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_inv_color.osl]: The code can be simplified very significantly:

    out_color_trans = in_color_trans;

    if (as_invert_channel_0 == 1)
         out_color_trans[0] = 1 - in_color_trans[0];

    if (as_invert_channel_1 == 1)
         out_color_trans[1] = 1 - in_color_trans[1];

    if (as_invert_channel_2 == 1)
         out_color_trans[2] = 1 - in_color_trans[2];

File: src/appleseed.shaders/src/appleseed/as_inv_color.osl [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    {
+        out_color_trans[0] = 1 - in_color_trans[0];
+        out_color_trans[1] = 1 - in_color_trans[1];
+        out_color_trans[2] = 1 - in_color_trans[2];
+    }
+    else
+    {
+        out_color_trans = in_color_trans;
+    }
+    out_outColor =  transformc(in_color_model, "rgb", out_color_trans);
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_inv_color.osl]: Remove extra space after `=`.

File: src/appleseed.shaders/src/appleseed/as_inv_color.osl (hunk 1 above)
Comment (Mango-3) [reply src/appleseed.shaders/src/appleseed/as_inv_color.osl]: ok

File: src/appleseed.shaders/src/appleseed/as_inv_color.osl (hunk 2 above)
Comment (Mango-3) [reply src/appleseed.shaders/src/appleseed/as_inv_color.osl]: I tried a simplified logic initially. But after testing I saw it didn't work when selecting 2 or more channels as inverted. So each case is explicitly stated, may look overly complicated but it works reliably. OSL is different from C++ and not all things work equally, e.g. there is no bool data type in OSL.

File: src/appleseed.shaders/src/appleseed/as_inv_color.osl (hunk 3 above)
Comment (Mango-3) [reply src/appleseed.shaders/src/appleseed/as_inv_color.osl]: done

**Comment by Mango-3** [comment]
Thanks for the speedy review!

File: src/appleseed.shaders/src/appleseed/as_inv_color.osl (hunk 2 above)
Comment (dictoon) [reply src/appleseed.shaders/src/appleseed/as_inv_color.osl]: I'm not sure to understand. OSL and C++ are different but still, valid OSL code will behave as expected, and the code I wrote above looks valid to me, so it should do the expected thing.

> But after testing I saw it didn't work when selecting 2 or more channels as inverted.

What happened? Are you sure your code was correct? Did you try mine?

Comment (Mango-3) [reply src/appleseed.shaders/src/appleseed/as_inv_color.osl]: You are right. I must have had a bug initially when I tested the simplified version the first time. Your simplification works as expected.

**Pull request merged**